src/plugins/stake-plugins/generic/util/stakeKitUtils.ts

```diff
@@ -63,3 +71,10 @@ export const yieldGetSingleYieldBalances = async (integrationId: string, yieldBa
 export const yieldYieldOpportunity = async (integrationId: string): Promise<YieldDto> => {
   return await fetchGet<YieldDto>(`/v1/yields/${integrationId}`)
 }
+
+const checkInsufficientFunds = (res: any): void => {
+  const maybeStakeKitInsufficientFundsError = asMaybe(asObject({ message: asValue('InsufficientFundsError') }))(res)
+  if (maybeStakeKitInsufficientFundsError != null) {
+    throw new InsufficientFundsError({ tokenId: null })
```

Comment:
What about making the `tokenId` a parameter? Then you would be doing `checkInsufficientFunds(out, null)` on the lines above. Unless you know for sure there there will never be tokens in this subsystem - then the current way is fine.

Reply:
The plugin is for Coreum native token staking only. If we start adding other StakeKit plugins we should add this for sure